core/services/relay/evm/functions/config_poller_test.go

```diff
@@ -0,0 +1,209 @@
+	db := pgtest.NewSqlxDB(t)
+	defer db.Close()
+	cfg := pgtest.NewQConfig(false)
+	ethClient := evmclient.NewSimulatedBackendClient(t, b, big.NewInt(1337))
+	defer ethClient.Close()
+	lggr := logger.TestLogger(t)
+	ctx := testutils.Context(t)
+	lorm := logpoller.NewORM(big.NewInt(1337), db, lggr, cfg)
+	lp := logpoller.NewLogPoller(lorm, ethClient, lggr, 100*time.Millisecond, 1, 2, 2, 1000)
+	defer lp.Close()
+	require.NoError(t, lp.Start(ctx))
+	logPoller, err := functions.NewFunctionsConfigPoller(pluginType, lggr, lp, ocrAddress)
+	require.NoError(t, err)
+	// Should have no config to begin with.
+	_, config, err := logPoller.LatestConfigDetails(testutils.Context(t))
+	require.NoError(t, err)
+	require.Equal(t, ocrtypes2.ConfigDigest{}, config)
+
+	pluginConfig := &functionsConfig.ReportingPluginConfigWrapper{
+		Config: &functionsConfig.ReportingPluginConfig{
+			MaxQueryLengthBytes:       10000,
+			MaxObservationLengthBytes: 10000,
+			MaxReportLengthBytes:      10000,
+			MaxRequestBatchSize:       10,
+			DefaultAggregationMethod:  functionsConfig.AggregationMethod(0),
+			UniqueReports:             true,
+			ThresholdPluginConfig: &functionsConfig.ThresholdReportingPluginConfig{
+				MaxQueryLengthBytes:       10000,
+				MaxObservationLengthBytes: 10000,
+				MaxReportLengthBytes:      10000,
+				RequestCountLimit:         100,
+				RequestTotalBytesLimit:    100000,
+				RequireLocalRequestCheck:  true,
+			},
+		},
+	}
+
+	// Set the config
+	contractConfig := SetFunctionsConfig(t, pluginConfig, ocrContract, user)
+	b.Commit()
+	latest, err := b.BlockByNumber(testutils.Context(t), nil)
+	require.NoError(t, err)
+	// Ensure we capture this config set log.
+	require.NoError(t, lp.Replay(testutils.Context(t), latest.Number().Int64()-1))
+
+	// Send blocks until we see the config updated.
+	var configBlock uint64
+	var digest [32]byte
+	gomega.NewGomegaWithT(t).Eventually(func() bool {
+		b.Commit()
+		configBlock, digest, err = logPoller.LatestConfigDetails(testutils.Context(t))
+		require.NoError(t, err)
+		return ocrtypes2.ConfigDigest{} != digest
+	}, testutils.WaitTimeout(t), 100*time.Millisecond).Should(gomega.BeTrue())
+
+	// Assert the config returned is the one we configured.
+	newConfig, err := logPoller.LatestConfig(testutils.Context(t), configBlock)
+	require.NoError(t, err)
+
+	// Get actual configDigest value from contracts
+	configFromContract, err := ocrContract.LatestConfigDetails(nil)
+	require.NoError(t, err)
+	onChainConfigDigest := configFromContract.ConfigDigest
+
+	assert.Equal(t, contractConfig.Signers, newConfig.Signers)
+	assert.Equal(t, contractConfig.Transmitters, newConfig.Transmitters)
+	assert.Equal(t, contractConfig.F, newConfig.F)
+	assert.Equal(t, contractConfig.OffchainConfigVersion, newConfig.OffchainConfigVersion)
+	assert.Equal(t, contractConfig.OffchainConfig, newConfig.OffchainConfig)
+
+	var expectedConfigDigest [32]byte
+	copy(expectedConfigDigest[:], onChainConfigDigest[:])
+	switch pluginType {
```

Comment:
It's a little silly that we copy-paste the same code that we're supposed to test. Maybe add "expectedPrefix" (byte[2]) parameter to runTest()?